### ISSUE TITLE
`useForm` hook doesn't work with `Checkbox.Card` the same way it works with `Checkbox`

### ISSUE THREAD

**Issue by spotlightsoftware** [mantinedev/mantine#7054]
### Dependencies check up

- [X] I have verified that I use latest version of all @mantine/* packages

### What version of @mantine/* packages do you have in package.json?

7.13.4

### What package has an issue?

@mantine/core

### What framework do you use?

Vite

### In which browsers you can reproduce the issue?

Chrome

### Describe the bug

Hi, I am using the `Checkbox.Card` component from the docs page.  I am trying to wire it up to the useForm hook, just like I did with a regular `Checkbox`, but it doesn't work (I can't check or uncheck it).  Am I missing something?

```jsx
<Checkbox.Card key={form.key("c1")} {...form.getInputProps('c1', { type: "checkbox" })} className={classes.root} radius="md">
    <Group className={classes.group} wrap="nowrap" align="flex-start">
        <Checkbox.Indicator />
        <div>
            <Text className={classes.label}>c1</Text>
            <Text className={classes.description}>c1 desc</Text>
        </div>
    </Group>
</Checkbox.Card>
```

Working checkbox:
```
<Checkbox key={form.key("c1")} {...form.getInputProps('c1', { type: "checkbox" })} label="c1" />
```

Thanks!

### Self-service

- [ ] I would be willing to implement a fix for this issue

**Comment by rtivital** [comment]
It is intended behavior, Checkbox.Card component is not an input, it does not have `onChange` event, instead the value changes is handled with `onClick` event. To use it with use form, do the following:

```tsx
<Checkbox.Card key={form.key("c1")} value={form.getValues().c1} onClick={() => ferm.setFieldValue('c1', !form.getValues().c1)} />
```

**Comment by cfrerebeau** [comment]
I don't think the documentation is clear on that point then. Thank you

**Comment by rtivital** [comment]
The issue is resolved in 7.14.3, Checkbox.Card component now work with form.getInputProps.
